File: API/Entities/Message.cs
namespace API.Entities;

public class Message
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public required string Content { get; set; }
    public DateTime MsgSentOn { get; set; } = DateTime.UtcNow;
    public DateTime? MsgReadOn { get; set; }
    public bool IsSenderMsgDeleted { get; set; }
    public bool IsRecipientMsgDeleted { get; set; }

    //Nav Properties

    public required string SenderId { get; set; }
    public Member Sender { get; set; } = null!;
    public required string RecipientId { get; set; }
    public Member Recipient { get; set; } = null!;
}

File: client/src/types/message.ts
export type Message = {
  id: string
  senderId: string
  senderDisplayName: string
  senderImgUrl: string
  recipientId: string
  recipientDisplayName: string
  recipientImgUrl: string
  content: string
  msgSentOn: string
  msgReadOn?: string
  currentUserSender?: boolean
}